pkg/api/types.go

```diff
@@ -1052,6 +1054,8 @@ type DependencyOverrides map[string]string
 type MultiStageTestConfigurationLiteral struct {
 	// ClusterProfile defines the profile/cloud provider for end-to-end test steps.
 	ClusterProfile ClusterProfile `json:"cluster_profile"`
+	// ClusterClaim claims an OpenShift cluster and exposes environment variable ${KUBECONFIG} to the test container
+	ClusterClaim *ClusterClaim `json:"cluster_claim,omitempty"`
```

Comment:
This is the problem: this field does not belong here.  `cluster_claim` is a field at the test level (https://docs.ci.openshift.org/docs/architecture/ci-operator#testing-with-a-cluster-from-a-cluster-pool), not at the multi-stage test definition level (i.e. the `steps` field).

`MultiStageTestConfiguration` should have a cluster claim field to enable what you are proposing (we can live with the asymmetry, IMO), but the `*Literal` counterpart should not.  No multi-stage test definition should have a `cluster_claim` (the confusion here in the tests is proof).  We'll likely need to adjust the resolution interface, since it is currently strictly `MultiStageTestConfiguration` → `*Literal`.